### PR TITLE
Prevent Sales Order Agent from skipping emails moved into monitored folders

### PR DESCRIPTION
**Pull request by tomasevicst**
## What & why

Stop advancing **Earliest Sync At** after email polling. Moved emails retain their original received timestamp, so advancing this value could permanently exclude them from later retrieval.

The configured Start Date remains the retrieval lower bound, while Graph excludes emails already categorized as processed. Added regression tests for empty and under-full polling.

## Linked work

Fixes [AB#648911](https://dynamicssmb2.visualstudio.com/Dynamics%20SMB/_workitems/edit/648911)

## How I validated this

- [x] I read the full diff and it contains only changes I intended.
- [x] I built the affected app(s) locally with no new analyzer warnings.
- [x] I ran the change in Business Central and confirmed it behaves as expected.
- [ ] I added or updated tests for the new behavior, or explained below why none are needed.

**What I tested and the outcome** *(required — be specific: scenarios, commands, screenshots for UI changes)*

## Risk & compatibility

Low risk. No schema, API, setup, or connector changes. Existing Start Date behavior during setup, reactivation, and mailbox changes remains unchanged. Previously processed emails continue to be excluded server-side by their Outlook category.

### CODE REVIEW

File: src/Apps/W1/SalesOrderAgent/app/src/Integration/SOARetrieveEmails.Codeunit.al [hunk 1]
```diff
@@ -93,7 +93,6 @@ codeunit 4582 "SOA Retrieve Emails"
         end;
 
         AddEmailInboxToSOAEmails(SOASetup, EmailInbox);
-        UpdateSOAEarliestSyncAt(SOASetup, EmailInbox.Count());
```
[the remark this answers is not in the document]
Comment (tomasevicst) [reply src/Apps/W1/SalesOrderAgent/app/src/Integration/SOARetrieveEmails.Codeunit.al:96]: The intent is to keep the configured Start Date as a fixed lower bound when monitoring a folder.

Emails moved into the folder retain their original **receivedDateTime**. Advancing **Earliest Sync At** after a successful poll can therefore permanently exclude emails that are moved into the folder later.

Already processed messages do not consume the retrieval batch: the connector sends the processed-category exclusion as part of the Microsoft Graph **$filter**, which Graph applies before **$top**. Keeping the Start Date unchanged trades a broader server-side query window for correctness and does not repeatedly return categorized messages. **Last Sync At** still records successful synchronization, but it cannot safely be used as the retrieval watermark for folder monitoring.

[the remark this answers is not in the document]
Comment (tomasevicst) [reply src/Apps/W1/SalesOrderAgent/app/src/Integration/SOARetrieveEmails.Codeunit.al:96]: This removal is intentional and is the accompanying design change.

The previous incremental-sync contract is invalid for folder monitoring: when an email is moved into the monitored folder, its original **receivedDateTime** is retained. Advancing **Earliest Sync At** therefore causes older emails moved later to be permanently skipped.

**Earliest Sync At** is the user-configured Start Date, not the actual synchronization progress. **Last Sync At** continues to record successful runs.

Deduplication is performed server-side: the Outlook connector includes the processed-category exclusion in the Microsoft Graph **$filter**, before **$top=50** is applied. Therefore, processed messages are not repeatedly returned or counted toward the batch.

A fixed lower bound does create a broader Graph query over time, but restoring the watermark trades correctness for performance and reintroduces the reported data-loss scenario. If query performance becomes problematic, Graph delta tracking would be the appropriate bounded solution. The PR includes regression tests confirming that empty and under-full polls preserve the configured Start Date.

**Comment by alexei-dobriansky** [comment]
## Good Sense Reviewer - Round 1

# **Recommendation: Accept with Suggestions**

<details>
<summary><strong>What this PR does</strong></summary>

This change stops advancing the Sales Order Agent email retrieval lower bound after each folder poll. That lets a message still be found if it was received earlier but moved into the monitored folder after a previous empty poll. Processed messages are still excluded by the existing Outlook category filter, so the change relies on that category as the duplicate guard.
</details>

<details>
<summary><strong>Problem-solution fit</strong></summary>

**Fit:** Strong

The reported problem is a race between folder moves and a received-time polling watermark. Removing the runtime watermark update directly addresses that race, and the surrounding retrieval still filters by folder and excludes messages already marked as processed.
</details>

<details>
<summary><strong>Suggestions</strong></summary>

**S1** (🟠 Moderate): *Add folder polling regression coverage*
The changed path removes the only runtime update of the retrieval lower bound, but this diff does not include a test for that behavior. Please add a test for an empty or under-limit folder poll, or explain why the connector mock cannot cover it.
</details>

<details>
<summary><strong>Risk assessment and necessity</strong></summary>

**Risk:** The regression surface is folder-based email polling in Sales Order Agent. The main risk is repeated scanning from the configured lower bound, but the folder filter, max-email limit, and processed category still bound normal processing and duplicate handling.

**Necessity:** The change is needed because moved messages keep their original received time. Without this change, a successful empty poll can permanently exclude an eligible message from later retrieval.
</details>

---
> [AI-PR-REVIEW] version=1 promptVersion=4 system=github pr=11063 round=1 by=alexei-dobriansky at=2026-09-04T18:15:21.3645173Z lastSha=b4243ea664021c78f129280cf662a2328b7fc19c reviewKey=a2d7df95cdbc83488abc15b99b58305851d0451ad3f6613c32c9a7a9d9e997d2 suggestions=S1@1f0e1b8e

**Review by PredragMaricic — APPROVED** [review]
The change correctly preserves the configured Start Date as the retrieval lower bound, preventing emails moved into a monitored folder after receipt from being permanently excluded. Processed messages remain excluded server-side by category, and the change is narrowly scoped.

**Review by ChethanT — APPROVED** [review]
